<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>994</width>
    <height>744</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Nexus Processing</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_7">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <widget class="QGroupBox" name="pulseInformation">
       <property name="title">
        <string>Pulse Information</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QGroupBox" name="spectraGroupBox">
          <property name="title">
           <string>Relevant Spectra Range</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QSpinBox" name="lowerSpecSpinBox"/>
           </item>
           <item>
            <widget class="QSpinBox" name="upperSpecSpinBox"/>
           </item>
           <item>
            <widget class="QPushButton" name="updateSpectraButton">
             <property name="text">
              <string>Update</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="pulsePlotLayout">
          <item>
           <widget class="SpectraTable" name="spectraTableView">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
           </widget>
          </item>
          <item>
           <widget class="EventTable" name="eventTableView">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_9">
       <item>
        <widget class="QGroupBox" name="periodDefinitionGroupBox">
         <property name="title">
          <string>Period Definition</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <layout class="QHBoxLayout" name="periodDefinitionLayout">
             <item>
             <widget class="QWidget" name="pulseTableGroup">
             <layout class="QHBoxLayout" name="boxLayout">
              <item>
               <widget class="PulseTable" name="pulseTableView">
                <property name="minimumSize">
                 <size>
                  <width>305</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="selectionBehavior">
                 <enum>QAbstractItemView::SelectRows</enum>
                </property>
               </widget>
              </item>
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_13">
                <item>
                 <widget class="QToolButton" name="addPulseButton">
                  <property name="text">
                   <string/>
                  </property>
                  <property name="icon">
                   <iconset>
                    <normaloff>:/icons/plus.png</normaloff>:/icons/plus.png</iconset>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QToolButton" name="removePulseButton">
                  <property name="text">
                   <string/>
                  </property>
                  <property name="icon">
                   <iconset>
                    <normaloff>:/icons/minus.png</normaloff>:/icons/minus.png</iconset>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="verticalSpacer_3">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>20</width>
                    <height>40</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
              </layout>
              </widget>
              </item>
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_10">
                <property name="sizeConstraint">
                 <enum>QLayout::SetDefaultConstraint</enum>
                </property>
                <property name="leftMargin">
                 <number>6</number>
                </property>
                <item>
                 <widget class="QRadioButton" name="usePeriodDefinitionsButton">
                  <property name="minimumSize">
                   <size>
                    <width>300</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="text">
                   <string>Use Period Definitions</string>
                  </property>
                  <property name="checked">
                   <bool>true</bool>
                  </property>
                  <attribute name="buttonGroup">
                   <string notr="true">buttonGroup_2</string>
                  </attribute>
                 </widget>
                </item>
                <item>
                 <widget class="QRadioButton" name="useAllPulsesButton">
                  <property name="text">
                   <string>Use All Pulses</string>
                  </property>
                  <attribute name="buttonGroup">
                   <string notr="true">buttonGroup_2</string>
                  </attribute>
                 </widget>
                </item>
                <item>
                <widget class="QWidget" name="periodDefinitionsGroup">
                <layout class="QVBoxLayout" name="verticalLayout8">
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_7">
                  <item>
                   <widget class="QLabel" name="label_4">
                    <property name="sizePolicy">
                     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                      <horstretch>0</horstretch>
                      <verstretch>0</verstretch>
                     </sizepolicy>
                    </property>
                    <property name="minimumSize">
                     <size>
                      <width>150</width>
                      <height>0</height>
                     </size>
                    </property>
                    <property name="text">
                     <string>Start Pulse Label</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QComboBox" name="pulseLabelComboBox">
                    <property name="minimumSize">
                     <size>
                      <width>0</width>
                      <height>0</height>
                     </size>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_8">
                  <item>
                   <widget class="QLabel" name="label">
                    <property name="toolTip">
                     <string>Extrapolate from current pulse</string>
                    </property>
                    <property name="text">
                     <string>Extrapolation</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QComboBox" name="extrapolationModeComboBox">
                    <property name="enabled">
                     <bool>true</bool>
                    </property>
                    <property name="minimumSize">
                     <size>
                      <width>0</width>
                      <height>0</height>
                     </size>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_10">
                  <item>
                   <widget class="QLabel" name="label_8">
                    <property name="toolTip">
                     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Slices to split master pulse in to (if FORWARDS_SUMMED)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                    </property>
                    <property name="text">
                     <string>Slices</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QSpinBox" name="slicingSpinBox">
                    <property name="minimum">
                     <number>1</number>
                    </property>
                    <property name="maximum">
                     <number>1000</number>
                    </property>
                    <property name="value">
                     <number>1</number>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_4">
                  <item>
                   <widget class="QLabel" name="label_2">
                    <property name="enabled">
                     <bool>true</bool>
                    </property>
                    <property name="minimumSize">
                     <size>
                      <width>0</width>
                      <height>0</height>
                     </size>
                    </property>
                    <property name="text">
                     <string>Period Duration</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="ExponentialSpinBox" name="periodDurationSpinBox">
                    <property name="maximum">
                     <double>10000000.000000000000000</double>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <spacer name="verticalSpacer_2">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>20</width>
                    <height>40</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                </layout>
                </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="runGroupBox">
         <property name="title">
          <string>Run</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QRadioButton" name="useTempDirButton">
            <property name="text">
             <string>Use temporary directory</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">buttonGroup</string>
            </attribute>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="useDataFileDirButton">
            <property name="text">
             <string>Use writable data file directory</string>
            </property>
            <attribute name="buttonGroup">
             <string notr="true">buttonGroup</string>
            </attribute>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="interpolateCheckBox">
            <property name="text">
             <string>Interpolate final data?</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Good frame threshold</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QSpinBox" name="goodFrameThresholdSpinBox">
              <property name="maximum">
               <number>1000000</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_12">
            <property name="topMargin">
             <number>10</number>
            </property>
            <item>
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>Output Directory</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_6">
              <item>
               <widget class="QLineEdit" name="outputDirLineEdit">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="browseOutputDirButton">
                <property name="text">
                 <string>Browse</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QPushButton" name="runButton">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Run</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SpectraTable</class>
   <extends>QTableView</extends>
   <header>src.gui.widgets.tables.spectra_table</header>
  </customwidget>
  <customwidget>
   <class>EventTable</class>
   <extends>QTableView</extends>
   <header>src.gui.widgets.table.event_table</header>
  </customwidget>
  <customwidget>
   <class>ExponentialSpinBox</class>
   <extends>QDoubleSpinBox</extends>
   <header>src.gui.widgets.exponential_spinbox</header>
  </customwidget>
  <customwidget>
   <class>PulseTable</class>
   <extends>QTableView</extends>
   <header>src.gui.widgets.tables.pulse_table</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources/resources.qrc"/>
 </resources>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroup"/>
  <buttongroup name="buttonGroup_2"/>
 </buttongroups>
</ui>
